docs/07-lifecycle.md

```diff
@@ -2,8 +2,68 @@
+
+AWS Lambda functions have semantic names, which means that a function described in the manifest.json file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example a name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. App developer does not need to worry about the AWS lambda function names, Proxy Plugin takes care of it.
+Dedicated S3 bucket name is stored in the environment variable 
+
+`MM_APPS_S3_BUCKET`
+
+which stores all apps' static assets and manifest files. 
+
+
+All files in the static folder of the bundle are considered to be the app's static assets. And are stored in the above mentioned bucket. Stored assets also have semantic keys and are generated using the rule - `static/$appID_$appVersion/filename`. For example `servicenow` app's static file key can be `"static/com.mattermost.servicenow_0.1.0_app/photo.png"`. App developer does not need to worry about the static asset keys, Proxy Plugin takes care of it.
+
+
+The `manifest.json` file of an app is stored in the same S3 bucket as well with the key - `manifests/$appID_$appVersion.json`.
+
+![](imgs/provisioning-in-3rd-party-aws.png)
+
+
+### Provisioning In Mattermost AWS Cloud
+To be provisioned in AWS mattermost cloud an app bundle is uploaded to the specific S3 bucket(`mattermost-apps-bucket`). On a new app release, bundle is created by the circleCI and uploaded to S3. [Mattermost apps cloud deployer](https://github.com/mattermost/mattermost-apps-cloud-deployer), running as a k8s cron job every hour, detects the S3 upload, creates appropriate lambda functions, assets and manifest the same way the **appsclt** does for the third party accounts. The deployer needs lambda function names, asset keys and manifest key to provision the app. It calls the `aws.GetProvisionDataFromFile(/PATH/TO/THE/APP/BUNDLE)` from the Proxy Plugin to get the provision data. Same data can be generated using the
+`go run ./cmd/appsctl/ generate-terraform-data /PATH/TO/YOUR/APP/BUNDLE` 
+![](imgs/provisioning-in-mm-aws.png)
+
+
 ## Publish
+publishing or registering an app in a Mattermost installation means the app will be shown in the Marketplace of the installation and it can be later installed by the sysadmin and used by the users. On a totally new app registration or on a registration of the new version of the already registered app, a new version of the Proxy Plugin is cut. `manifests.json` file is updated and a new app is added in the listing. Later, plugin is installed in the appropriate installations(using feature flags if necessary). After the plugin update Proxy Plugin synchronizes the list of the registered apps by downloading appropriate manifests from the S3 and storing them in memory. So the Marketplace shows renewed app listings and sysadmin can install a new app(or new version).
```

Comment:
Let's clean up the naming here. What does `Publish` mean vs. `Register`? Is that is the same? Or is one about the status of an app release and the other about the process?

Reply:
I'll leave it to @justinegeffen